unit/java_bytecode/java_bytecode_parse_generics/parse_functions_with_generics.cpp

```diff
@@ -0,0 +1,340 @@
+/*******************************************************************\
+
+ Module: Unit tests for parsing generic classes
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+
+#include <util/config.h>
+#include <util/cmdline.h>
+#include <util/language.h>
+#include <util/prefix.h>
+
+#include <java_bytecode/java_bytecode_language.h>
+
+#include <iostream>
+#include <testing-utils/load_java_class.h>
+#include <testing-utils/require_type.h>
+
+SCENARIO(
+  "java_bytecode_parse_functions_with_generics",
+  "[core][java_bytecode][java_bytecode_parse_generics]")
+{
+  const symbol_tablet &new_symbol_table =
+    load_java_class(
+      "FunctionsWithGenerics", "./java_bytecode/java_bytecode_parse_generics");
+
+  std::string class_prefix = "java::FunctionsWithGenerics";
+
+  WHEN("Parsing processReturnSame")
+  {
+    THEN("There should be a symbol for the function")
+    {
+      const std::string func_name = ".processReturnSame";
+      const std::string func_descriptor = ":(LGeneric;)LGeneric;";
+      const std::string process_func_name =
+        class_prefix + func_name + func_descriptor;
+
+      REQUIRE(new_symbol_table.has_symbol(process_func_name));
+
+      const symbolt func_symbol =
+        new_symbol_table.lookup_ref(process_func_name);
+      const code_typet func_code = to_code_type(func_symbol.type);
```

Comment:
Maybe use `require_code` here instead of `to_code_type` ? Or possibly even add a `require_code_symbol` to `testing_utils/require_symbol.cpp` ?